codec/src/main/java/io/netty/handler/codec/ReplayingDecoder.java

```diff
@@ -418,7 +418,7 @@ protected void callDecode(ChannelHandlerContext ctx, ByteBuf in, List<Object> ou
             }
         } catch (DecoderException e) {
             throw e;
-        } catch (Throwable cause) {
+        } catch (Exception cause) {
```

Comment:
based on the javadoc of ReplayingDecoder I think we might need special handling here like we now have for Sni.. https://github.com/jasontedor/netty/blob/8ad05f00703ed42480e4b83c5d550dce68259b60/codec/src/main/java/io/netty/handler/codec/ReplayingDecoder.java#L77

```
* {@link ReplayingDecoder} passes a specialized {@link ByteBuf}
 * implementation which throws an {@link Error} of certain type when there's not
 * enough data in the buffer.
```

Reply:
@jasontedor https://github.com/netty/netty/blob/eb7f751ba519cbcab47d640cd18757f09d077b55/common/src/main/java/io/netty/util/Signal.java

Reply:
actually it's caught further up, might be okay.

Reply:
@johnou yep I think its fine